cpp/include/cuml/fil/fil.h

```diff
@@ -33,7 +33,8 @@ namespace fil {
 enum algo_t {
   /** choose the algorithm automatically; currently chooses NAIVE for sparse forests 
       and BATCH_TREE_REORG for dense ones */
-  ALGO_AUTO,
+  AUTO,
```

Comment:
The reason this constant is named `ALGO_AUTO` is that `AUTO` didn't compile for me (there's already an `AUTO` in `storage_type_t`). Does it compile now?

Reply:
my bad, it doesn't. And assuming conversion to enum class has too much of a chance to break a customer's C/C++ code?
in which case, elaborating the error message is all I can think of

Reply:
@JohnZed Do we have any particular objections to `enum class`?